File: fibonacci.erl
-module(fibonacci).
-export([fibonacci/1, main/0]).

fibonacci(N) when N =< 1 ->
    N;
fibonacci(N) ->
    fibonacci(N - 1) + fibonacci(N - 2).

main() ->
    io:format("~p~n", [fibonacci(10)]),
    io:format("~p~n", [fibonacci(20)]),
    io:format("~p~n", [fibonacci(30)]),
    io:format("~p~n", [fibonacci(35)]),
    io:format("~p~n", [fibonacci(40)]).

File: fibonacci.java
// javac fibonacci.java
// time java fibonacci

public class fibonacci {
  public static int fibonacci(int n) {
    if (n <= 1) {
      return n;
    } else {
      return fibonacci(n - 1) + fibonacci(n - 2);
    }
  }

  public static void main(String[] args) {
    System.out.println(fibonacci(10));
    System.out.println(fibonacci(20));
    System.out.println(fibonacci(30));
    System.out.println(fibonacci(35));
  }
}
